### PR TITLE
feat(manager/github-actions): extract complex github tags

### PR DESCRIPTION
**Pull request by Yokann**
### Github Action manager
Removing version control from the extractor allows more tuning with `packageRules`
It should be the responsibility of the versioning updater to check the version's validity.

## Context

Github Action extractor skipped entries when we tried to use tags with a prefix like: 
```
uses: monorepo/actions/workflows/deployment.yaml@deployment-v1
uses: monorepo/actions/workflows/tests.yaml@test-v1.2
```

I want Renovate to be allowed to upgrade to version `deployment-v2` with this kind of package rule: 
```
packageRules: [
    {
      matchDatasources: ["github-tags"],
      matchPackageNames: ["monorepo/actions"],
      versioning: "docker",
      versionCompatibility: "^(?<compatibility>.*)-v(?<version>.*)$",
    },
  ]
```

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

**Comment by jycamier** [comment]
Any news about this ?

File: lib/modules/manager/github-actions/extract.ts [hunk 1]
```diff
@@ -98,11 +101,11 @@ function extractWithRegex(content: string): PackageDependency[] {
...
         dep.currentValue = tag;
         dep.currentDigestShort = currentValue;
+      } else if (customVersioningRe.test(currentValue)) {
+        dep.currentValue = currentValue;
       } else {
         dep.currentValue = currentValue;
-        if (!dockerVersioning.api.isValid(currentValue)) {
-          dep.skipReason = 'invalid-version';
-        }
+        dep.skipReason = 'invalid-version';
```
Comment (rarkins) [inline lib/modules/manager/github-actions/extract.ts:108]: Why this change?

Comment (Yokann) [reply lib/modules/manager/github-actions/extract.ts:108]: This "docker...isValid" part is not relevant anymore because `customVersioningRe` will do the same 
```
} else if (customVersioningRe.test(currentValue)) {
        dep.currentValue = currentValue;
``` 
I should rename this variable.

Comment (viceice) [reply lib/modules/manager/github-actions/extract.ts:108]: I think we should accept any value and leave it to later versioning to decide if it's a valid version.

that way we can support almost all version schemes.

Comment (Yokann) [reply lib/modules/manager/github-actions/extract.ts]: This PR is more simple now :)
If everything is ok, I will edit the PR description.

File: lib/modules/manager/github-actions/extract.spec.ts [hunk 2]
```diff
@@ -402,8 +401,10 @@ describe('modules/manager/github-actions/extract', () => {
           currentValue: 'actions-runner-controller-0.23.5',
           replaceString:
             'actions-runner-controller/execute-assert-arc-e2e@f1d7c52253b89f0beae60141f8465d9495cdc2cf # actions-runner-controller-0.23.5',
-        },
+        }
```
Comment (viceice) [inline lib/modules/manager/github-actions/extract.spec.ts:404]:
```suggestion
        },
```

**Review by viceice — APPROVED** [review]
